server/server.go

```diff
@@ -200,6 +202,9 @@ func (s *server) Run(rootCtx context.Context) (err error) {
 	if s.network != nil {
 		stopCtx, stopCtxCancel := context.WithTimeout(context.Background(), StopOnSignalTimeout)
 		defer stopCtxCancel()
+		// we need to cancel the Start context so that any pending action with that context
+		// can be cancelled
+		s.startCtxCancel()
```

Comment:
as there are two ways to call network.stop() here (server receiving SIGINT or server.Stop() call), maybe this start context creation can be moved down to network.start() and the start context cancelation can be moved down to network.stop() so network struct manages that logic

Reply:
If I understood you correctly here this is a good suggestion. Changed.